src/climbingAPI/app/routers/pitch.py

```diff
@@ -0,0 +1,132 @@
+    # created ascent
+    created_ascent = await db["ascent"].find_one({"_id": new_ascent.inserted_id})
+    if created_ascent is None:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail=f"Ascent {new_ascent.inserted_id} not found")
+    # ascent was found
+    update_result = await db["pitch"].update_one({"_id": ObjectId(pitch_id)}, {"$push": {"ascent_ids": new_ascent.inserted_id}})
+    if update_result.modified_count != 1:
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                            detail=f"Ascent {new_ascent.inserted_id} was not added to pitch {pitch_id}")
+    # ascent_id was added to pitch
+    return JSONResponse(status_code=status.HTTP_201_CREATED,
+                        content=jsonable_encoder(AscentModel(**created_ascent)))
+
+
+@router.delete('/{pitch_id}/ascent/{ascent_id}', description="Delete an ascent", response_model=PitchModel, dependencies=[Depends(auth.implicit_scheme)])
+async def delete_ascent(pitch_id: str, ascent_id: str, user: Auth0User = Security(auth.get_user, scopes=["write:diary"])):
+    db = await get_db()
+    pitch = await db["pitch"].find_one({"_id": ObjectId(pitch_id), "user_id": user.id})
+    if pitch is None:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail=f"Pitch {pitch_id} not found")
+    # pitch was found
+    ascent = await db["ascent"].find_one({"_id": ObjectId(ascent_id), "user_id": user.id})
+    if ascent is None:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail=f"Ascent {ascent_id} not found")
+    # ascent was found
+    if await db["pitch"].find_one({"_id": ObjectId(pitch_id), "ascent_ids": ObjectId(ascent_id)}) is None:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT,
+                            detail=f"Ascent {ascent_id} does not belong to pitch {pitch_id}")
+    # ascent belongs to pitch
+    delete_result = await db["ascent"].delete_one({"_id": ObjectId(ascent_id)})
+    if delete_result.deleted_count != 1:
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                            detail=f"Ascent {ascent_id} could not be deleted")
+    # ascent was deleted
+    update_result = await db["pitch"].update_one({"_id": ObjectId(pitch_id)}, {"$pull": {"ascent_ids": ObjectId(ascent_id)}})
+    if update_result.modified_count != 1:
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                            detail=f"Removing ascent_id {ascent_id} from pitch {pitch_id} failed")
+    # ascent_id was removed
+    if (updated_pitch := await db["pitch"].find_one({"_id": ObjectId(pitch_id)})) is not None:
+        return updated_pitch
+    else:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND,
+                            detail=f"Pitch {pitch_id} not found")
+
```

Comment:
What is the reason to have the delete and create function for an ascent in the pitch.py router file?